File: HWLesson7/app/src/main/java/com/example/hwlesson7/NoteDetailFragment.java
package com.example.hwlesson7;

// NoteDetailFragment.java

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

public class NoteDetailFragment extends Fragment {
    private static final String ARG_TITLE = "title";
    private static final String ARG_DESC = "description";
    private static final String ARG_DATE = "date";

    public static NoteDetailFragment newInstance(Note note) {
        NoteDetailFragment fragment = new NoteDetailFragment();
        Bundle args = new Bundle();
        args.putString(ARG_TITLE, note.getTitle());
        args.putString(ARG_DESC, note.getDescription());
        args.putString(ARG_DATE, note.getDate());
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_note_detail, container, false);
        TextView titleView = view.findViewById(R.id.detail_title);
        TextView descView = view.findViewById(R.id.detail_description);
        TextView dateView = view.findViewById(R.id.detail_date);

        if (getArguments() != null) {
            titleView.setText(getArguments().getString(ARG_TITLE));
            descView.setText(getArguments().getString(ARG_DESC));
            dateView.setText(getArguments().getString(ARG_DATE));
        }
        return view;
    }
}

File: HWLesson6/app/src/main/java/com/example/hwlesson6/NoteDetailFragment.kt
package com.example.hwlesson6

import android.app.DatePickerDialog
import android.icu.util.Calendar
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.core.content.ContentProviderCompat.requireContext
import androidx.fragment.app.Fragment

class NoteDetailFragment(private val note: Note) : Fragment() {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return inflater.inflate(R.layout.fragment_note_detail, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val title = view.findViewById<EditText>(R.id.editTitle)
        val desc = view.findViewById<EditText>(R.id.editDescription)
        val textDate = view.findViewById<TextView>(R.id.textDate)
        val button = view.findViewById<Button>(R.id.buttonPickDate)

        title.setText(note.title)
        desc.setText(note.description)
        textDate.text = note.date

        button.setOnClickListener {
            val calendar = Calendar.getInstance()
            DatePickerDialog(requireContext(), { _, y, m, d ->
                textDate.text = "%02d.%02d.%d".format(d, m + 1, y)
            }, calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH)).show()
        }
    }
}
